packages/fxa-settings/src/components/App/index.tsx

```diff
@@ -3,36 +3,78 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 import React from 'react';
-import AccountDataHOC from '../AccountDataHOC';
-import { AccountData } from '../AccountDataHOC/gql';
+import { gql, useQuery } from '@apollo/client';
 import AppLayout from '../AppLayout';
+import LoadingSpinner from 'fxa-react/components/LoadingSpinner';
+import AppErrorDialog from 'fxa-react/components/AppErrorDialog';
 import Settings from '../Settings';
 import AppErrorBoundary from 'fxa-react/components/AppErrorBoundary';
 import { QueryParams } from '../../lib/types';
 import FlowEvents from '../../lib/flow-event';
+import { Account } from '../../models';
+
+export const GET_INITIAL_STATE = gql`
+  query GetInitialState {
+    account {
+      uid
+      displayName
+      avatarUrl
+      accountCreated
+      passwordCreated
+      recoveryKey
+      emails {
+        email
+        isPrimary
+        verified
+      }
+      attachedClients {
+        clientId
+        isCurrentSession
+        userAgent
+        deviceType
+        deviceId
+      }
+      totp {
+        exists
+        verified
+      }
+      subscriptions {
+        created
+        productName
+      }
+    }
+    session {
+      verified
+    }
+  }
+`;
 
 type AppProps = {
   queryParams: QueryParams;
 };
 
 export const App = ({ queryParams }: AppProps) => {
+  const { loading, error, data } = useQuery<{ account: Account }>(
+    GET_INITIAL_STATE
+  );
   FlowEvents.init(queryParams);
 
+  if (loading) {
+    return (
+      <LoadingSpinner className="bg-grey-20 flex items-center flex-col justify-center h-screen select-none" />
+    );
+  }
+
+  if (error) {
+    return <AppErrorDialog data-testid="error-dialog" {...{ error }} />;
+  }
+  const account = data!.account;
```

Comment:
we won't even care about `data` in this component once `Settings` uses `useAccount`